File: sangjin/level1/24_소수_찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> prime(1000001, 1);
    
    for(int i = 2; i <= prime.size()/2; i++){
        for(int j = 2; i*j < prime.size(); j++){
            prime[i*j] = 0;
        }
    }

    for(int i = 2; i <= n; i++){
        if(prime[i] == 1){
            answer++;
        }
    }

    return answer;
}

File: yunj/level1/24_소수 찾기.java
class Solution {
    public int solution(int n) {
          int[] prime = new int[n+1];

        for(int i = 2; i<= n ; i++){
            prime[i] = 1;

            int last = (int)Math.sqrt(i * 1.0);
            for(int j = 2; j <= last; j++){
                if (i % j == 0){
                    prime[i]=0;
                    break;
                }
            }

            prime[i] += prime[i-1];
        }
        return prime[n];
    }
}